### PR TITLE
feat: tokenchip implementation

### LINKED ISSUE #11: [DappKit] Token Chip

### PR DESCRIPTION
**Pull request by metalboyrick**
Fixes #11 

- Add `TokenChip` implementation with minimal deps based on https://onchainkit.xyz/token/token-chip
- Add example storybooks
- Added documentation and how-tos

### CODE REVIEW

**Comment by metalboyrick** [comment]
@debuggingfuture @justin-lau will we be implementing component level unit tests?

**Comment by justin-lau** [comment]
> @debuggingfuture @justin-lau will we be implementing component level unit tests?

It’s a good idea. Please feel free to start with this one.

File: apps/storybook/src/stories/onchain-info/TokenChip.mdx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+- [Button](https://ui.shadcn.com/docs/components/button)
+
+Copy the following code into your code base. And modify import parts as needed.
+
+```tsx
+import { Button } from "@/components/ui/button"; // modify as needed
+
+export type TokenChipProps = {
+  // optional because native tokens don't have addresses
```
Comment (justin-lau) [inline apps/storybook/src/stories/onchain-info/TokenChip.mdx:31]: Is this comment describing the attributes that exist on the Token type from onchainkit? If so it doesn't seem necessary here.

File: apps/storybook/src/stories/onchain-info/TokenChip.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+} satisfies Meta<typeof TokenChip>;
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const ETHTokenChip: Story = {
+  args: {
+    image:
+      "https://ethereum.org/_next/image/?url=%2F_next%2Fstatic%2Fmedia%2Feth-diamond-black.a042df77.png&w=828&q=75",
```
Comment (debuggingfuture) [inline apps/storybook/src/stories/onchain-info/TokenChip.stories.tsx:22]: lets curate a fixture of images of tokens

we might find those on ipfs / upload ourselves

**Review by debuggingfuture — CHANGES_REQUESTED** [review]

File: apps/storybook/src/stories/onchain-info/TokenChip.mdx (hunk 1 above)
Comment (metalboyrick) [reply apps/storybook/src/stories/onchain-info/TokenChip.mdx:31]: Was thinking if we re expecting any autofetches in the future, if not  i will delete no prob

File: apps/storybook/src/stories/onchain-info/TokenChip.stories.tsx (hunk 2 above)
Comment (metalboyrick) [reply apps/storybook/src/stories/onchain-info/TokenChip.stories.tsx:22]: Oke ser

File: apps/storybook/src/stories/onchain-info/TokenChip.mdx (hunk 1 above)
Comment (justin-lau) [reply apps/storybook/src/stories/onchain-info/TokenChip.mdx:31]: Could you clarify what you meant by autofetches?

Comment (justin-lau) [reply apps/storybook/src/stories/onchain-info/TokenChip.mdx:31]: @metalboyrick saw you pushed a commit to remove the comment on another file. Could you remove it on this file as well?

Comment (metalboyrick) [reply apps/storybook/src/stories/onchain-info/TokenChip.mdx]: fixed

File: apps/storybook/src/stories/onchain-info/TokenChip.stories.tsx (hunk 2 above)
Comment (metalboyrick) [reply apps/storybook/src/stories/onchain-info/TokenChip.stories.tsx]: this will be for later as per discussed

**Comment by metalboyrick** [comment]
Adding additional config to support DOM checking with vitest.

File: apps/storybook/src/stories/onchain-info/__tests__/TokenChip.test.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		expect(symbolText).toBeDefined();
+	});
+
+	it("renders with no image when image prop is not provided", () => {
+		render(<TokenChip name="Bitcoin" symbol="BTC" />);
+
+		// Check that the symbol and name are rendered correctly without image
+		const symbolText = screen.getByText("BTC");
+		expect(symbolText).toBeDefined();
+	});
```
Comment (justin-lau) [inline apps/storybook/src/stories/onchain-info/__tests__/TokenChip.test.tsx:46]: Based on the description, I think this test case should explicitly checks for the non-existence of the image.

**Review by justin-lau — CHANGES_REQUESTED** [review]

File: apps/storybook/src/stories/onchain-info/__tests__/TokenChip.test.tsx (hunk 3 above)
Comment (metalboyrick) [reply apps/storybook/src/stories/onchain-info/__tests__/TokenChip.test.tsx:46]: got it, currently the `img` tag would still be there even if link is `undefined`, will just add a check to not render it if link does not exist + write the corresponding test cases.

**Review by justin-lau — APPROVED** [review]

**Pull request merged**